pythonforandroid/toolchain.py

```diff
@@ -583,7 +586,7 @@ def add_parser(subparsers, *args, **kwargs):
             logger.setLevel(logging.DEBUG)
 
         self.ctx = Context()
-        self.ctx.use_setup_py = args.use_setup_py
+        self.ctx.use_setup_py = getattr(args, "use_setup_py", True)
```

Comment:
I think it would be neater to move the --use-setup-py etc. arguments to generic_parser so that they're always in the args object, instead of using getattr in different places. It isn't a great api to have to stick non-generic stuff in generic_parser like this, but I think it's a failure of the way the abstraction is set up rather than something specific to this particular argument. There are plenty of other things in generic_parser for the same reason.

Reply:
I disagree, I think it shouldn't be in the generic_parser if it's not for the other commands. I'd rather have the special cases that are admittedly ugly but still be able to understand to which commands the arguments belong, BOTH as a coder and a user. Moving it into generic_parser seems like a cheap hack to me

However, as usual if you really think that's how it should be I will change it accordingly, so your call :+1: